src/slice-assign.h

```diff
@@ -1,24 +1,48 @@
 #ifndef VCTRS_SLICE_ASSIGN_H
 #define VCTRS_SLICE_ASSIGN_H
 
+enum vctrs_ownership {
+  vctrs_ownership_owned,
```

Comment:
Should it be `vctrs_ownership_total`?

The first case is also the default case (= 0), it should be unknown.